rclcpp_generic/test/rclcpp_generic/test_pubsub.cpp

```diff
@@ -0,0 +1,188 @@
+// Copyright 2018, Bosch Software Innovations GmbH.
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+#include <gmock/gmock.h>
+
+#include <future>
+#include <memory>
+#include <string>
+#include <vector>
+
+#include "rclcpp/rclcpp.hpp"
+#include "rclcpp/serialization.hpp"
+
+#include "memory_management.hpp"
+
+#include "test_msgs/message_fixtures.hpp"
+#include "test_msgs/msg/basic_types.hpp"
+
+#include "rclcpp_generic/generic_publisher.hpp"
+#include "rclcpp_generic/generic_subscription.hpp"
+
+using namespace ::testing;  // NOLINT
+using namespace rclcpp_generic;  // NOLINT
+
+class RosBag2NodeFixture : public Test
+{
+public:
+  RosBag2NodeFixture()
```

Comment:
You'll probably want to do a find-and-replace for "Rosbag2" naming, and call it something like `RclcppGeneric` instead